### PR TITLE
Aggregate bench project

### PR DESCRIPTION
**Pull request by regadas**
With this we propagate `compile` and `fmt` tasks/commands.

### CODE REVIEW

**Comment by regadas** [comment]
hm, need to dig a little bit more as this runs locally? 🤔

**Comment by regadas** [comment]
🙌 alright so I guess this is now doing the expected.

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -55,7 +55,10 @@ jobs:
       - name: Check that workflows are up to date
         run: sbt ++${{ matrix.scala }} githubWorkflowCheck
 
-      - run: sbt ++${{ matrix.scala }} fmtCheck test mimaReportBinaryIssues
+      - run: |
+          sbt ++${{ matrix.scala }} fmtCheck \
```
Comment (johnynek) [inline .github/workflows/ci.yml:59]: why do we need this change? Also, it is interesting there is no quote on the first item.

File: build.sbt [hunk 2]
```diff
@@ -26,7 +26,13 @@ ThisBuild / githubWorkflowPublishTargetBranches := Seq(
 )
 
 ThisBuild / githubWorkflowBuild := Seq(
-  WorkflowStep.Sbt(List("fmtCheck", "test", "mimaReportBinaryIssues"))
+  WorkflowStep.Run(
+    List(
+      """sbt ++${{ matrix.scala }} fmtCheck \
```
Comment (johnynek) [inline build.sbt:31]: I guess this is the difference, note the fmtCheck doesn't have quotes around it... But does it need the `++` at all? The format check does not depend on the scala version does it?

**Review by johnynek — COMMENTED** [review]
Thanks for dealing with this. I know it can be a frustration to deal with the build.

File: build.sbt (hunk 2 above)
Comment (regadas) [reply build.sbt:31]: > The format check does not depend on the scala version does it?

selecting the scala version on format will allow us to check the format on scala version specific code i.e under`scala-2.1x` folders. Currently we don't have any so we don't really need it. I thought of separating the "checks" into a separate task and run it for one version only but ended up keeping the previous behaviour; can be handy? why not?

The quotes are an interesting case and one of the weirdest behaviours in SBT. When running sbt like this(non interactive) Scala version is not correctly selected on projects when running tasks ... quoting them helps. However, the behavior is not the same on cmds like `fmtCheck` and leads to error when it expands.

Note: from sbt console you don't need the quotes.

Comment (johnynek) [reply build.sbt:31]: Thanks for the detailed reply.

**Review by johnynek — APPROVED** [review]

**Pull request merged**